[Joven Pua Zai Xiong] iP

CSmortal opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

CSmortal committed:
Level 1 1st change

CSmortal committed:
Level 2: 1st change

CSmortal committed:
Level 3: 1st change

CSmortal committed:
added Task.java for level 3

CSmortal committed:
changed Duke.java to work with Task.java: 2nd change to Level 3

CSmortal committed:
added Todo.java

CSmortal committed:
added Deadline.java

CSmortal committed:
added Event.java

CSmortal committed:
changed Duke.java for Level 4

CSmortal committed:
added DukeException.java

CSmortal committed:
added some error handling

CSmortal committed:
removed unnecessary println()

CSmortal committed:
changed for Level-6

CSmortal committed:
minor changes, not important at all

CSmortal committed:
added DukeCommand enum

CSmortal committed:
abstracted most println() in Duke to a logging function

CSmortal committed:
add save functionality

CSmortal committed:
add fileFormat method

CSmortal committed:
updated save functionality

CSmortal committed:
modify Deadline and Event to store dates as LocalDate instead

CSmortal committed:
modify Duke.java to work with LocalDate for Deadline and Event

CSmortal committed:
Merge branch 'branch-Level-7'

CSmortal committed:
remove println in Duke's delete handler

CSmortal committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java
#              src/main/java/Deadline.java
#              src/main/java/Event.java

CSmortal committed:
manage import statements in Duke

CSmortal committed:
add additional classes for OOP refactoring

CSmortal committed:
update Duke for OOP refactoring

CSmortal committed:
move all classes into new package

CSmortal committed:
add ParserTest, update Duke and Parser

CSmortal committed:
add Javadoc comments

CSmortal committed:
add find functionality

CSmortal committed:
Merge branch 'branch-A-JavaDoc'

CSmortal committed:
Merge branch 'branch-Level-9'

CSmortal committed:
edit for code style

CSmortal committed:
Merge branch 'branch-A-CodingStandard'

CSmortal committed:
modify package statement

CSmortal committed:
add getters and equals()

CSmortal committed:
edit testcases

ZechariahTan commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,74 @@
+        if (obj == this) {
+            return true;
+        } else {
+            return obj instanceof Deadline && description.equals(((Deadline) obj).getDescription())
+                    && deadline.equals(((Deadline) obj).getDeadline()) && isDone == ((Deadline) obj).isDone();
```
Could this expression be expanded? It looks a little difficult to decipher. ([Textbook section](https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions))

ZechariahTan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,108 @@
+                } else if (dukeCommand.getCommand() == Command.INVALID) {
+                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-()");
+
+                } else if (dukeCommand.getCommand() == Command.FIND) {
+                    if (dukeCommand.getDetails().length() == 0) {
+                        throw new DukeException("☹ OOPS!!! Search keyword cannot be empty");
+                    }
+
+                    ListIterator<Task> taskIter = taskList.getTasks().listIterator();
+                    String keyword = dukeCommand.getDetails();
+                    TaskList matchedTasks = new TaskList();
+
+                    while (taskIter.hasNext()) {
+                        Task curr = taskIter.next();
+
+                        if (curr.getDescription().contains(keyword)) {
+                            matchedTasks.add(curr);
+                        }
+                    }
+
+                    Ui.showMatchedTasks(matchedTasks);
+
+
+                } else {
```
Could these blocks be simplified / abstracted out? ([Textbook section](https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent))

ZechariahTan commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,74 @@
+     * @return a new Deadline that is considered done
+     */
+    @Override
+    public Deadline markAsDone() {
```
I like that all your method names are verbs where applicable

ZechariahTan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,149 @@
+        try {
+            String[] params = input.split(" ", 2);
+
+            if (input.equals("bye")) {
+                return new DukeCommand(Command.BYE, "");
+
+            } else if (params[0].equals("delete")) {
+                return new DukeCommand(Command.DELETE, params[1]);
+
+            } else if (input.equals("list")) {
+                return new DukeCommand(Command.LIST, "");
+
+            } else if (params[0].equals("done")) {
+                return new DukeCommand(Command.DONE, params[1]);
+
+            } else if (params[0].equals("todo")) {
+                return new DukeCommand(Command.TODO, params[1]);
+
+            } else if (params[0].equals("deadline")) {
+                return new DukeCommand(Command.DEADLINE, params[1]);
+
+            } else if (params[0].equals("event")) {
+                return new DukeCommand(Command.EVENT, params[1]);
+
+            } else if (params[0].equals("find")) {
+                return new DukeCommand(Command.FIND, params[1]);
+            } else {
+                return new DukeCommand(Command.INVALID, "");
```
Would a switch case here make the code more readable? I don't think its in the coding standard so it might just be a personal opinion but the block of code here looks like it fits exactly what a switch case would do.

ZechariahTan reviewed (commented):
Looks like it follows most standard coding practices. Just needs simplifying some areas for readability's sake

CSmortal committed:
add DukeExceptionTest

CSmortal committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jared98lyj commented on src/main/java/Duke.java:
```diff
@@ -1,6 +1,6 @@
 import main.java.*;
```
No wildcard imports

jared98lyj commented:
Generally good and succinct code. Code could be simplified in some parts, e.g. using enhanced for loop instead of ListIterator. Can also store repeated access of lists to a variable, e.g. currentCommand = params[0]; Using more representative names would also help. action/command over input as a variable name.

CSmortal committed:
add gradle support

CSmortal committed:
edit to conform to checkstyle

CSmortal committed:
Merge branch 'A-Checkstyle'

CSmortal committed:
replace text UI with a GUI

CSmortal committed:
Merge branch 'branch-Level-10'

CSmortal committed:
edit mainClassName to duke.Launcher

CSmortal committed:
refactor main code

CSmortal committed:
add relevant assertions

CSmortal committed:
Improve code quality

Updated javadoc comments, simplify exception handling.

Task objects now throw DukeException if markAsDone() is
called on them when they are already done

CSmortal committed:
edit some assert statements to AssertionError

CSmortal committed:
Merge pull request #2 from CSmortal/branch-A-Assertions

Branch a assertions

CSmortal committed:
revert merge pull request

CSmortal committed:
revert 2 commits back to commit e08791c

CSmortal committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/utility/Parser.java
#	src/main/java/utility/Storage.java
#	src/main/java/utility/Ui.java

CSmortal committed:
Merge pull request #3 from CSmortal/branch-A-CodeQuality

Improve code quality

CSmortal committed:
Add tagging functionality (add, delete)

CSmortal committed:
Merge pull request #4 from CSmortal/branch-C-Tagging

Add tagging functionality (add, delete)

CSmortal committed:
improve GUI

CSmortal committed:
Add representative screenshot to /docs

CSmortal committed:
Fix tag parsing and included testing for tags

CSmortal committed:
Merge branch 'fixTagParsing'

CSmortal committed:
refactor code to make it more OOP

CSmortal committed:
Fix testcases and update javadoc comments

CSmortal committed:
Merge branch 'Refactor-Duke-for-SLAP-and-OOP'

CSmortal committed:
update README
